### PR TITLE
DDT-672: PUT response payload

### PR DESCRIPTION
**Pull request by Abdiiir**
- updated body response & unit tests

### CODE REVIEW

File: src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java
Comment (douradofrazer) [inline src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:705]: I did the same fix in a PR before this that was merged, it's weird that it does not conflict.

Comment (Abdiiir) [reply src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:705]: Yeah, I looked into this and can see your point. Not sure why this happened, but the code is not duplicated

**Review by douradofrazer — DISMISSED** [review]

File: src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java
Comment (douradofrazer) [reply src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:705]: @Abdiiir make sure you verify that the code isn't overwritten on merge.

Comment (Abdiiir) [reply src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java]: This should be resolved now and the duplicate line caused from re-basing is removed. 

@douradofrazer waiting for a second review.

File: src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java
Comment (mp-assco) [inline src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:691]: I think we need some checks similar to what we have at the beginning of `createBooking` concerning `getIsImportLicenseRequired`, `getImportLicenseReference`, `getIsExportDeclarationRequired`, `getExportDeclarationReference`.

Comment (Abdiiir) [reply src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:691]: Yes, will include that.

File: src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java
Comment (douradofrazer) [inline src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:697]: This looks like duplicate code, extract it out ?

File: src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java
Comment (douradofrazer) [inline src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:1534]: I think we can consolidate this with the above method and keep the check on only `allowedDocumentStatuses` I don't think there is a specific need to check for `CANC`, @pimg what do you think?

**Review by douradofrazer — CHANGES_REQUESTED** [review]

File: src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java
Comment (Abdiiir) [reply src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:1534]: Do we not then somehow allow for cancelled bookings to be updated?

Comment (HenrikHL) [reply src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:1534]: I am quite sure Christian has said that once a Booking is Cancelled - then it cannot be modified... @pimg please also confirm...

Comment (pimg) [reply src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:1534]: correct cancelled is an end state

**Review by douradofrazer — APPROVED** [review]
I do see some duplications in checks for update and post while reviewing the PR's, Will be a good idea to refactor once merged.

**Pull request merged**
